pkg/engine/planner/physical/optimizer.go

```diff
@@ -155,13 +155,13 @@ type projectionPushdown struct {
 func (r *projectionPushdown) apply(node Node) bool {
 	switch node := node.(type) {
 	case *VectorAggregation:
-		if len(node.GroupBy) == 0 || node.Operation != types.VectorAggregationTypeSum {
+		if len(node.GroupBy) == 0 || !slices.Contains(types.SupportedVectorAggregationTypes, node.Operation) {
```

Comment:
I think we should be careful with pushing down `groupBy` for vector aggregations.

For example `max by (level) (sum_over_time{...})` is not equivalent to `max by (level) (sum_over_time{...} by (level))`.
The first is looking at max value by level over **sum per series** while the second is looking over `sum per level` which is bound to be much bigger.